src/routes/url.ts

```diff
@@ -42,15 +44,13 @@ const retrieveUrl: Route<{ Params: { id: string }; Querystring: { redirect?: str
 	attachValidation: true,
 	async handler(request, reply) {
 		if (request.validationError) throw new NotFoundError()
+		const withInfo = await this.auth.isAuthorized(request)
+		const storedUrl = await this.storage.url.get(request.params.id, { withInfo })
+		if (typeof storedUrl === 'undefined') throw new NotFoundError()
 
-		const url = await this.storage.url.get(request.params.id)
-		if (typeof url === 'undefined') throw new NotFoundError()
+		this.storage.url.incInfoCount(request.params.id)
```

Comment:
Probably a noob question, but - is there any chance incVisitCount won't complete successfully since we're not waiting for it to end before returning the request and closing the connection?